tests/testthat/test-public_api.R

```diff
@@ -19,5 +19,36 @@ test_that("styler can style file", {
 test_that("styler does not return error when there is no file to style", {
   expect_error(style_dir(paste0(base, "/xyzemptydir"), strict = FALSE), NA)
 })
+context <- structure(list(
+  id = "C533793B",
+  path = paste0(base, "/xyzaddin/addin_region-in.R"),
+  contents = c("fjkdsfa 2jy+wj/ 1+1 <?+d", ""),
+  selection = structure(list(structure(list(range = structure(list(
+    start = structure(c(1, 17), .Names = c("row", "column"), class = "document_position"),
+    end = structure(c(1, 21), .Names = c("row", "column"),
+                    class = "document_position")), .Names = c("start", "end"),
+    class = "document_range"), text = "1+1 "), .Names = c("range", "text"))),
+    .Names = "", class = "document_selection")),
+  .Names = c("id", "path", "contents", "selection"), class = "document_context"
+)
 
+# style_active_file() must be tested manually.
+test_that("styling active region works", {
 
+  before_styling <- utf8::read_lines_enc(paste0(base, "/xyzaddin/addin_region-in.R"))
```

Comment:
Maybe just copy to a temporary directory?

```r
dir <- tempfile("styler")
dir.create(dir)
file.copy(".../addin_region-out.R", dir)
```

The current approach seems slightly brittle if the tests aborts before the original contents have been written.

Reply:
Ok, right. So if I understand you right, we copy the `*-in.R*` to the temporary directory, style it there and compare it with the `-*out`. in the permanent directory?

Reply:
Yes.